### PR TITLE
enh(CPlusPlus): add missing space to power-of-two maths

### PR DESCRIPTION
**Pull request by darshan-trivedi-10**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [ ] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] New algorithm
- [x] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
How to check power of two using Binary Method in O(1) Time Complexity.

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Maths/power-of-two.cpp [hunk 1]
```diff
@@ -1,40 +1,50 @@
...
     return i == n;
 }
 
-//main starts
+// Binary Method to check number is in power of two or not.
+// Time Complexity of this method is O(1)
+bool ispoweroftwo(const int n)
+{
+    // if number is in power of two it will give 0, otherwise it will not zero.
+    return (n & (n - 1)) == 0;
```
Comment (Ashborn-SM) [inline algorithms/CPlusPlus/Maths/power-of-two.cpp:24]: what do you think the result will be if you give `n = 0` for this function?

**Review by Ashborn-SM — CHANGES_REQUESTED** [review]

File: algorithms/CPlusPlus/Maths/power-of-two.cpp (hunk 1 above)
Comment (Ashborn-SM) [inline algorithms/CPlusPlus/Maths/power-of-two.cpp:24]: what do you think the result will be if you give n = 0 for this function?

Comment (darshan-trivedi-10) [reply algorithms/CPlusPlus/Maths/power-of-two.cpp:24]: Yes, I forgot this case sorry i will add this case.

Comment (darshan-trivedi-10) [reply algorithms/CPlusPlus/Maths/power-of-two.cpp]: @Ashborn-SM  change code. kindly review it.

File: algorithms/CPlusPlus/Maths/power-of-two.cpp [hunk 2]
```diff
@@ -1,40 +1,50 @@
...
 }
 
 /*
 Example 1:
 
 Input: n = 1
-Output:1 (true) 
+Output:1 (true)
 Explanation: 20 = 1
 Example 2:
```
Comment (Ashborn-SM) [inline algorithms/CPlusPlus/Maths/power-of-two.cpp:44]:
```suggestion
Output:1 (true)
Explanation: 2^0 = 1

Example 2:
```

File: algorithms/CPlusPlus/Maths/power-of-two.cpp [hunk 3]
```diff
@@ -1,40 +1,50 @@
...
 
 Input: n = 16
-Output:1 (true) 
+Output:1 (true)
 Explanation: 24 = 16
 Example 3:
```
Comment (Ashborn-SM) [inline algorithms/CPlusPlus/Maths/power-of-two.cpp:49]:
```suggestion
Output:1 (true)
Explanation: 2^4 = 16

Example 3:
```

**Review by Ashborn-SM — CHANGES_REQUESTED** [review]

**Review by Ashborn-SM — APPROVED** [review]
Good work!!

**Pull request merged**